pkg/registry/apis/provisioning/files.go

```diff
@@ -206,6 +209,10 @@ func (s *filesConnector) doWrite(ctx context.Context, update bool, repo reposito
 		return nil, err
 	}
 
+	if !isJSONContentType(req) {
```

Comment:
writing yaml also works...

the request will also fail if it does not parse as a Resource, so i don't think this is necessary

Reply:
I vote we remove this check -- it would already fail (but not as quickly)

Reply:
what about checking additional if it's YAML content type?

Reply:
no strong feeling... but it just feels extra since we are checking that it parses as yaml|json anyway